ui/tests/integration/components/replication-secondary-card-test.js

```diff
@@ -0,0 +1,90 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+const REPLICATION_DETAILS = {
+  state: 'stream-wals',
+  connection_state: 'ready',
+  lastWAL: 0,
+  lastRemoteWAL: 10,
+  delta: 10,
+};
+
+const STATE_ERROR = {
+  state: 'idle',
+  connection_state: 'ready',
+  lastWAL: 0,
+  lastRemoteWAL: 10,
+  delta: 10,
+};
+
+const CONNECTION_ERROR = {
+  state: 'idle',
+  connection_state: 'transient-failure',
+  lastWAL: 0,
+  lastRemoteWAL: 10,
+  delta: 10,
+};
+
+module('Integration | Enterprise | Component | replication-secondary-card', function(hooks) {
+  setupRenderingTest(hooks);
+  const title = 'States';
+  const hasErrorClass = false;
+
+  hooks.beforeEach(function() {
+    this.set('replicationDetails', REPLICATION_DETAILS);
+    this.set('stateError', STATE_ERROR);
+    this.set('connectionError', CONNECTION_ERROR);
+    this.set('hasErrorClass', hasErrorClass);
+    this.set('title', title);
+  });
+
+  test('it renders', async function(assert) {
+    await render(
+      hbs`<ReplicationSecondaryCard @replicationDetails={{replicationDetails}} @title={{title}} @hasErrorClass={{hasErrorClass}}/>`
+    );
+
+    assert.dom('[data-test-replication-secondary-card]').exists();
+  });
+  test('it renders with state, and connection set', async function(assert) {
+    await render(
+      hbs`<ReplicationSecondaryCard @replicationDetails={{replicationDetails}} @title={{title}} @hasErrorClass={{hasErrorClass}}/>`
+    );
+    assert.dom('[data-test-state]').includesText(REPLICATION_DETAILS.state, `shows the correct state value`);
+
+    assert
+      .dom('[data-test-connection]')
+      .includesText(REPLICATION_DETAILS.connection_state, `shows the correct connection value`);
+  });
+
+  test('it renders with lastWAL, lastRemoteWAL and delta set when title is not States', async function(assert) {
+    await render(
+      hbs`<ReplicationSecondaryCard @replicationDetails={{replicationDetails}} @hasErrorClass={{hasErrorClass}}/>`
+    );
+    assert
+      .dom('[data-test-lastWAL]')
+      .includesText(REPLICATION_DETAILS.lastWAL, `shows the correct lastWAL value`);
+    assert
+      .dom('[data-test-lastRemoteWAL]')
+      .includesText(REPLICATION_DETAILS.lastRemoteWAL, `shows the correct lastRemoteWAL value`);
+    assert.dom('[data-test-delta]').includesText(REPLICATION_DETAILS.delta, `shows the correct delta value`);
+  });
+
+  test('it renders tooltip with check-circle-outline when state is stream-wals', async function(assert) {
+    await render(
+      hbs`<ReplicationSecondaryCard @replicationDetails={{replicationDetails}} @title={{title}} />`
+    );
+    assert.dom('[data-test-glyph]').hasClass('has-text-success', `shows success icon`);
```

Comment:
similar to my previous comments i think this assertion could be moved into the first test case since there's nothing different here in the test set up. that is, the component rendered here has the same properties as the component rendered in the first test. when i read this as a separate test, i was scanning, looking for where we actually set the state to `stream-wals` but when i couldn't find it i realized there was nothing different. the `data-test-glyph` selector could be updated to be more clear too, maybe `data-test-tooltip`? 

hopefully that makes sense! i'm realizing that many of my comments here are more about testing theory vs. syntax, so let me know if you'd like to discuss this more holistically sometime. i could talk about this during one of our ember learn sessions, although this stuff isn't necessarily specific to ember. 😉

Reply:
These are great comments. I amended this to check with the specific error message and title.  Same with the previous tests.